model/task/task.go

```diff
@@ -3128,6 +3128,102 @@ func GetTaskStatsByVersion(versionID string, opts GetTasksByVersionOptions) ([]*
 	return StatusCount, nil
 }
 
+type GroupedStatusCount struct {
+	Variant      string         `bson:"variant"`
+	DisplayName  string         `bson:"display_name"`
+	StatusCounts []*StatusCount `bson:"status_counts"`
+}
+
+func GetGroupedTaskStatsByVersion(versionID string, opts GetTasksByVersionOptions) ([]*GroupedStatusCount, error) {
+	StatusCount := []*GroupedStatusCount{}
+	pipeline := getTasksByVersionPipeline(versionID, opts)
+	project := bson.M{"$project": bson.M{
+		BuildVariantKey:            "$" + BuildVariantKey,
+		BuildVariantDisplayNameKey: "$" + BuildVariantDisplayNameKey,
+		DisplayStatusKey:           "$" + DisplayStatusKey,
+	}}
+	pipeline = append(pipeline, project)
+	variantStatusesKey := "variant_statuses"
+	groupByStatusPipeline := []bson.M{
+		{
+			"$group": bson.M{
+				"_id": "$" + BuildVariantKey,
```

Comment:
Can we add some comments here to make it clear what's going on and what the different stages are for? (Something like creating a list here of all statuses per build variant, and then unwind/project moves variantStatusesKey to top level and removes the ID, and then we count?)